### ISSUE TITLE
ESLint isn't reporting unresolved modules

### ISSUE THREAD

**Issue by rhettlivingston** [meteor/todos#157]
While working on another update, I noted that ESLint isn't reporting unresolved modules in import statements as expected. I could actually change an import such as
```js
import { Lists } from '../lists/lists.js';

// to

import { Lists } from '../lists/junk-lists.js';
```
and still get a clean report from 'meteor npm run lint'.

After some investigation, I realized that this was due to outdated versions of the devDependencies which when corrected revealed a missing devDependency for eslint-import-resolver-meteor (in addition to one for eslint-plugin-jsx-a11y that is due to a change in eslint-plugin-react's requirements).

After correcting these dependencies, 'meteor npm run lint' reports quite a few problems, partly due to new rules in updated plugins and partly modules that should be in '.meteor/packages' but are only in '.meteor/versions' (i.e. we got away with the imports because a module we are importing them imported them for us).

```bash

```

59 out of the 71 errors are due to a new(ish) rule in airbnb, [no-underscore-dangle](http://eslint.org/docs/rules/no-underscore-dangle). Basically, "_" and "_id" are no longer allowed as variable names. We could create a list of those we'd like to allow in the rule configuration,,, but I think we should just disable it.

Other errors are due to unresolved paths to modules including
- meteor/useraccounts:core
- meteor/practicalmeteor:chai
- meteor/practicalmeteor:sinon
that just need to be 'meteor add'ed to the project.

That leaves 
```
/home/rhett/oob/meteors/todosfork/imports/api/generate-data.app-tests.js
  28:1  error  Exporting mutable 'let' binding, use 'const' instead  import/no-mutable-exports

/home/rhett/oob/meteors/todosfork/imports/api/lists/lists.tests.js
  16:3  error  Unexpected require()  global-require

/home/rhett/oob/meteors/todosfork/imports/api/todos/todos.tests.js
  14:3  error  Unexpected require()  global-require
```

I'm unsure of how to fix the import/no-mutable-exports error and can't fix the "global-require" errors without updating to 1.3.4.1 and using the new conditional imports. My thought is to disable these on those lines only. For now.

I'm submitting a PR shortly that implements all of the above.

**Comment by tmeasday** [comment]
Fixing the let could be done pretty easily with code like

```js
let generateDataClient;
if (Meteor.isClient) {
  ...
}

const generateData = Meteor.isClient ? generateDataClient : null;
```

**Comment by rhettlivingston** [comment]
Closed via #158.

**Comment by lexcaraig** [comment]
`
import React, {PropTypes} from 'react';
import { connect } from 'react-redux';
import { addTodo } from '../actions';

let AddTodo = ({ dispatch }) => {
  let input;

  return (
    <div>
      <form
        onSubmit={e => {
          e.preventDefault();
          if (!input.value.trim()) {
            return;
          }
          dispatch(addTodo(input.value));
          input.value = '';
        }}>
        <input
          ref={node => {
            input = node;
          }} />
        <button
          type="submit">
          Add Todo
        </button>
      </form>
    </div>
  );
};

AddTodo.propTypes = {
  dispatch: PropTypes.func.isRequired
};

AddTodo = connect()(AddTodo);

export default AddTodo;

`

![screenshot from 2016-09-27 10-44-14](https://cloud.githubusercontent.com/assets/15936424/18858544/5cca3376-849f-11e6-92c1-b1342673b4fa.png)


Hi, i got this error and I can't figure out what's happening, can you help me? Thanks!
